strings/truncate_string.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:test/test.dart';
+
+String truncateString(String string,
+    {int maxNumberOfWords, String suffix = '...'}) {
+  return (string.length - 3) < maxNumberOfWords
+      ? "$string$suffix"
+      : '${string.substring(0, string.substring(0, maxNumberOfWords - suffix.length).lastIndexOf(" "))}$suffix';
+}
+
+void main() {
+  test(
+    "Truncate a long string given the max number of words and or suffix",
+    () {
+      String stringToTruncate =
+          "The beautiful thing with learning is that no one can take it away from you";
+      expect(truncateString(stringToTruncate, maxNumberOfWords: 40),
+          equals("The beautiful thing with learning is..."));
+    },
+  );
```

Comment:
Add more test cases. More Strings and edge cases.

Reply:
Hello @akashgk! 
Added more test cases...
Thanks for reviewing